ssl/methods.c

```diff
@@ -118,26 +118,7 @@
 
 static const SSL_METHOD *tls1_get_method(int ver)
 {
-    if (ver == TLS_ANY_VERSION)
-        return TLS_method();
-#ifndef OPENSSL_NO_TLS1_2
-    if (ver == TLS1_2_VERSION)
-        return tlsv1_2_method();
-#endif
-#ifndef OPENSSL_NO_TLS1_1
-    if (ver == TLS1_1_VERSION)
-        return tlsv1_1_method();
-#endif
-#ifndef OPENSSL_NO_TLS1
-    if (ver == TLS1_VERSION)
-        return tlsv1_method();
-#endif
-#ifndef OPENSSL_NO_SSL3
-    if (ver == SSL3_VERSION)
-        return (sslv3_method());
-    else
-#endif
-    return NULL;
+    return TLS_method();
```

Comment:
Rather than no-op-ing the `get_ssl_method` call in `SSL_set_session`, I think it's better to remove it. This does some odd things for the version-locked methods. (Granted, the old code does odd things too, albeit different ones.)